packages/internal-test-env/index.ts

```diff
@@ -143,113 +141,147 @@ function getTestingEnvironment(
+      environment: process.env.E2E_TEST_ENVIRONMENT,
+      features: featuredFlags,
+    };
+  
+    return libVars ? merge(base, validateLibVars(libVars)) : base;
+}
+
 export function getNodeTestingEnvironment(
   libVars?: LibraryVariables
 ): TestingEnvironmentNode {
-  setupEnv();
-  getTestingEnvironment(process.env);
-
-  if (
-    process.env.E2E_TEST_REQUESTOR_CLIENT_ID !== undefined ||
-    process.env.E2E_TEST_REQUESTOR_CLIENT_SECRET !== undefined
-  ) {
-    if (typeof process.env.E2E_TEST_REQUESTOR_CLIENT_ID !== "string") {
-      throw new Error(
-        "The environment variable E2E_TEST_REQUESTOR_CLIENT_ID is undefined."
-      );
-    }
-
-    if (typeof process.env.E2E_TEST_REQUESTOR_CLIENT_SECRET !== "string") {
-      throw new Error(
-        "The environment variable E2E_TEST_REQUESTOR_CLIENT_SECRET is undefined."
-      );
-    }
-  }
-
-  const base = {
-    idp: process.env.E2E_TEST_IDP,
-    environment: process.env.E2E_TEST_ENVIRONMENT,
-    clientCredentials: {
-      owner: {
-        id: process.env.E2E_TEST_OWNER_CLIENT_ID,
-        secret: process.env.E2E_TEST_OWNER_CLIENT_SECRET,
-      },
-    },
-    features: featuredFlags,
-  };
+  return getBaseTestingEnvironment({
+    ...libVars,
+    // Enforce client credentials are present for the resource owner.
+    clientCredentials: { owner: { id: true, secret: true }}
+  });
+}
 
-  return libVars ? merge(base, validateLibVars(libVars)) : base;
+export function getBrowserTestingEnvironment(
+  libVars?: LibraryVariables
+): TestingEnvironmentBrowser {
+  return getBaseTestingEnvironment({
+    ...libVars,
+    // Enforce login/password are present for the resource owner.
+    clientCredentials: { owner: { login: true, password: true }}
+  });
 }
 
 export interface LibraryVariables {
-  notificationGateway?: string;
-  notificationProtocol?: string;
-  vcProvider?: string;
+  notificationGateway?: boolean;
+  notificationProtocol?: boolean;
+  vcProvider?: boolean;
   clientCredentials?: {
     owner?: {
-      id?: string;
-      secret?: string;
-      login?: string;
-      password?: string;
+      id?: boolean;
+      secret?: boolean;
+      login?: boolean;
+      password?: boolean;
     };
     requestor?: {
-      id?: string;
-      secret?: string;
+      id?: boolean;
+      secret?: boolean;
+    };
+  };
+}
+
+export interface BrowserVariables extends LibraryVariables {
+  clientCredentials?: {
+    owner: {
+      login: true;
+      password: true;
+    };
+  };
+}
+
+export interface NodeVariables extends LibraryVariables {
+  clientCredentials?: {
+    owner: {
+      id: true;
+      secret: true;
     };
   };
 }
 
 function validateLibVars(vars: LibraryVariables): object {
+  // Notifications-related environment variables.
   if (
-    typeof vars.notificationGateway !== "undefined" &&
-    typeof vars.notificationGateway !== "string"
+    vars.notificationGateway &&
+    typeof process.env.E2E_TEST_NOTIFICATION_GATEWAY !== "string"
```

Comment:
We probably also want a "isValidUrl" check here

Reply:
55034786daef0bb3d00361ed69ffb7b2f8527f6c